R/ae_table_soc.R

```diff
@@ -124,58 +127,86 @@ ae_table_soc = function(
     set_names(to_snake_case)
 
   extra_cols = if(total) c("NA", "Tot") else c("NA")
+
   rtn = df %>%
     summarise(calc = evaluate_grades(grade_, variant),
               .by=any_of(c("subjid_", "arm_", "soc_", "term_"))) %>%
     unnest(calc) %>%
     mutate(
-      # Tot = ifelse(total, Tot, 0),
       soc_ = soc_ %>% fct_infreq(w=Tot) %>%
-             fct_last(label_missing_soc, label_missing_pat)
-    ) %>%
+        fct_last(label_missing_soc, label_missing_pat)
+    )
+
+
+  if (!is.null(ae_groups) && is.list(ae_groups)) {
```

Comment:
Il faudrait ajouter de la programmation défensive ici :
Si `!is.null(ae_groups)`, on check que c'est une liste, sinon `cli_abort()`, on check que les valeurs sont des grades valides (1-5 ou NA) sinon `cli_abort()`, etc.